### PR TITLE
Fix PReP detection

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/RAPZuM0j/640-3-storageng-bootloader-bootloader-does-not-detect-prep-boot-partition.

### CODE REVIEW

File: src/modules/BootStorage.rb [hunk 1]
```diff
@@ -221,18 +221,9 @@ def reset_disks
...
-        parts.concat(pair[1]["partitions"] || [])
-      end
-
-      prep_partitions = partitions.select do |partition|
-        [0x41, 0x108].include? partition["fsid"]
-      end
-
-      y2milestone "detected prep partitions #{prep_partitions.inspect}"
-      prep_partitions.map { |p| p["device"] }
+      partitions = Y2Storage::Partitionable.all.map(&:prep_partitions).join
```
Comment (ancorgs) [inline src/modules/BootStorage.rb]: Shouldn't this be? `Y2Storage::Partitionable.all(staging).map(&:prep_partitions).flatten`

File: src/modules/BootStorage.rb [hunk 2]
```diff
@@ -221,18 +221,9 @@ def reset_disks
...
+      y2milestone "detected prep partitions #{partitions.inspect}"
```
Comment (ancorgs) [inline src/modules/BootStorage.rb]: A nice opportunity to change this into `log.info`

File: test/stage1_test.rb [hunk 3]
```diff
@@ -7,20 +7,16 @@
 Yast.import "Arch"
 Yast.import "BootStorage"
 
-xdescribe Bootloader::Stage1 do
+describe Bootloader::Stage1 do
   before do
     # simple mock getting disks from partition as it need initialized libstorage
     allow(subject).to receive(:can_use_boot?).and_return(true)
-    mock_disk_partition
+    # mock_disk_partition
```
Comment (ancorgs) [inline test/stage1_test.rb]: I would just kill the whole line

File: src/modules/BootStorage.rb (hunk 1 above)
Comment (joseivanlopez) [inline src/modules/BootStorage.rb]: Sure, thanks!

**Comment by ancorgs** [comment]
Can I suggest this test to be more sure?:
```ruby
describe Yast::BootStorage do
  subject { Yast::BootStorage }

  describe ".prep_partitions" do
    it "returns the correct set of Y2Storage::Partition objects" do
      devicegraph_stub("prep_partitions.yml")
      partitions = subject.prep_partitions
      expect(partitions).to all(be_a(Y2Storage::Partition))
      expect(partitions.map(&:name)).to contain_exactly("/dev/sda2", "/dev/sdb2")
    end 
  end 
```

With, for example, this as YAML (in `test/data/storage-ng/prep_partitions.yml`)
```yaml
- disk:
    size: 800 GiB
    name: "/dev/sda"
    partition_table: msdos
    partitions:

    - partition:
        size: 75 GiB
        name: /dev/sda1
        id: dos32
        file_system: vfat

    - partition:
        size: 40 GiB
        name: /dev/sda2
        id: prep
        file_system: vfat

    - partition:
        size: 2 GiB
        name: /dev/sda3
        id: dos16
        file_system: vfat

    - partition:
        size: unlimited
        name: /dev/sda4
        file_system: ext4

- disk:
    name: "/dev/sdb"
    size: 50 GiB
    partition_table: gpt
    partitions:
    - partition:
        size: 200 MiB
        name: "/dev/sdb1"
        type: primary
        id: linux
        file_system: ext4
        mount_point: "/boot"
    - partition:
        size: 1 MiB
        name: "/dev/sdb2"
        type: primary
        id: prep
    - partition:
        size: 43009 MiB (42.00 GiB)
        name: "/dev/sdb3"
        type: primary
        id: lvm
```

**Comment by ancorgs** [comment]
LGTM again after `push -f`

**Pull request merged**